### PR TITLE
Add APIs to set viewport and scissor rect for render pass

### PR DESCRIPTION
**Pull request by Richard-Yunchao**
PTAL. 

See the investigation about **Viewport and Scissor Test** at https://github.com/gpuweb/gpuweb/issues/120

### CODE REVIEW

**Comment by Richard-Yunchao** [comment]
@Kangz , @kainino0x , @kvark , @RafaelCintron , @litherum , PTAL when you have time.

File: design/sketch.webidl [hunk 1]
```diff
@@ -481,6 +481,14 @@ interface WebGPUProgrammablePassEncoder {
 interface WebGPURenderPassEncoder : WebGPUProgrammablePassEncoder {
     void setBlendColor(float r, float g, float b, float a);
     void setStencilReference(u32 reference);
+
+    // The default viewport is (0.0, 0.0, w, h, 0.0, 1.0), where w and h are the dimensions of back buffer
+    void setViewport(float x, float y, float, width, float height, float minDepth, float maxDepth);
+
+    // The default scissor rectangle is (0, 0, w, h), where w and h are the demensions of back buffer.
```
Comment (kvark) [inline design/sketch.webidl]: typo "demensions"

Comment (Richard-Yunchao) [reply design/sketch.webidl]: Done. Thanks for your review, @kvark

**Comment by kvark** [comment]
@Richard-Yunchao please squash the commits

**Comment by Richard-Yunchao** [comment]
> @Richard-Yunchao please squash the commits

Done. I remember that we can merge multiple commits into one commit when we merge the pr. So I didn't squash the commits before, @kvark . 

If you prefer to squash commits into one commit, I will do that in future.

**Comment by kvark** [comment]
@Richard-Yunchao you are right. I just got used to bad automerge bots that don't know how to squash :)

**Comment by Richard-Yunchao** [comment]
Can you take a look? @kvark . Shall we remove the comments for these two functions in this pr?

**Comment by litherum** [comment]
In the future we may want support for multiple viewports (e.g. for VR). We should keep that future direction in mind.

File: design/sketch.webidl [hunk 2]
```diff
@@ -481,6 +481,14 @@ interface WebGPUProgrammablePassEncoder {
 interface WebGPURenderPassEncoder : WebGPUProgrammablePassEncoder {
     void setBlendColor(float r, float g, float b, float a);
     void setStencilReference(u32 reference);
+
+    // The default viewport is (0.0, 0.0, w, h, 0.0, 1.0), where w and h are the dimensions of back buffer
+    void setViewport(float x, float y, float, width, float height, float minDepth, float maxDepth);
```
Comment (frzi) [inline design/sketch.webidl]: Rouge comma (`,`) between `float width`. :)

Comment (kainino0x) [reply design/sketch.webidl]: Fixed in prep for merge. Thanks for finding this!

**Comment by Kangz** [comment]
Aaaand @kainino0x fixed it, nice!

**Comment by Richard-Yunchao** [comment]
> In the future we may want support for multiple viewports (e.g. for VR). We should keep that future direction in mind.

Yes. I also simply discussed with a few guys about multiple viewports for WebVR several days ago. In addition, I added this statement into <Conclusion and Proposal> section at the investigation issue for Viewport and scissor tests. Thanks for your reminder, @litherum.

File: design/sketch.webidl (hunk 2 above)
Comment (Richard-Yunchao) [reply design/sketch.webidl]: Thanks!

**Pull request merged**